packages/next/src/server/next-server.ts

```diff
@@ -719,99 +722,108 @@ export default class NextNodeServer extends BaseServer {
     res: BaseNextResponse,
     parsedUrl: NextUrlWithParsedQuery
   ) {
-    if (this.minimalMode || this.nextConfig.output === 'export') {
+    if (
+      this.minimalMode ||
+      this.nextConfig.output === 'export' ||
+      process.env.NEXT_MINIMAL
+    ) {
       res.statusCode = 400
       res.body('Bad Request').send()
       return {
         finished: true,
       }
-    }
-    const { ImageOptimizerCache } =
-      require('./image-optimizer') as typeof import('./image-optimizer')
+    } else {
```

Comment:
There is a return above. I'm guessing it's intentionally `else` for minify shaking the condition? Can you add a comment indicating that otherwise it gets lost in a refactor.
```suggestion
    }
```